src/Infrastructure/Repositories/EmployeeRepository.cs

```diff
@@ -413,19 +413,74 @@ public async Task AddOrUpdateEmployeeInformation(string employeeEmail, Emergency
     public async Task<List<ProjectExperience>> GetProjectExperiencesByEmailAndCompetencies(string email,
         List<string> competencies)
     {
+        // Get the employee
         var employeeId = await _db.Employees
             .Where(emp => emp.Email == email)
             .Select(e => e.Id)
             .SingleOrDefaultAsync();
 
-        return await _db.ProjectExperiences
-            .Include(pe => pe.ProjectExperienceRoles)
-            .Include(pe => pe.Competencies)
-            .Where(pe =>
-                pe.EmployeeId == employeeId &&
-                _db.Competencies.Any(c => competencies.Contains(c.Name) && c.ProjectExperienceId == pe.Id))
-            .Select(pe => pe.ToProjectExperience())
-            .ToListAsync();
+        // Setup our ranking of competencies
+        List<(int, string)> competenciesLength = new List<(int, string)>();
+
+        // Calculate how long the employee has worked with each competency
+        foreach (var competence in competencies)
+        {
+            // Get project experiences for each competency
+            var projectExperiences = await _db.ProjectExperiences
+                .Include(pe => pe.ProjectExperienceRoles)
+                .Include(pe => pe.Competencies)
+                .Where(pe =>
+                    pe.EmployeeId == employeeId &&
+                    _db.Competencies.Any(c => competence.Contains(c.Name) && c.ProjectExperienceId == pe.Id))
+                .Select(pe =>  pe.ToProjectExperience())
+                .ToListAsync();
+
+            // Sum all project lengths for each competency
+            TimeSpan totalCompetenceExperience = projectExperiences.Aggregate(TimeSpan.Zero, (currentTotal, projectexpreience) => {
+                DateOnly fromDate = projectexpreience.FromDate ?? DateOnly.FromDateTime(DateTime.Now);
+                TimeSpan interval = projectexpreience.ToDate.ToDateTime(TimeOnly.MinValue) - fromDate.ToDateTime(TimeOnly.MinValue);
+                return currentTotal + interval;
+            });
+
+            // Set num years for each competency
+            competenciesLength.Add((((int)totalCompetenceExperience.TotalDays)/365, competence));

```

Comment:
Her runder vi av til heltall by the way. Er det ønskelig å dele på 365 her eller kan vi bruke dager videre?

Reply:
Det er bare fordi på trello står det at vi vil oppgi erfaring i antall år.

Reply:
Vi oppgir erfaring i år, men dette er vel mer et internt hjelpeverktøy. Må i så fall castet til en datatype som støtter desimaler.